torch/csrc/jit/codegen/cuda/lower_utils.cpp

```diff
@@ -612,6 +654,369 @@ ParallelTypeBitmap getParallelBroadcastDomains(
+    }
+
+    // Grab the axis information
+    auto ca_point = tv->getComputeAtAxis(tv_i);
+    auto ca_view = ca_point.second;
+    auto ca_id = ca_point.first;
+
+    // use ca_id map if we have it.
+    if (!ca_id_map.empty()) {
+      ca_id = ca_id_map.at(ca_id);
+    }
+
+    loops_it = std::find_if(loops_it, loops.end(), [&ca_id](const auto& loop) {
+      return ca_id == loop->iter_domain();
+    });
+
+    TORCH_INTERNAL_ASSERT(
+        loops_it != loops.end(),
+        "Could not find all required axes for indexing.");
+
+    ca_to_loop_map[ca_id] = *loops_it;
+    ++loops_it;
+  }
+
+  return ca_to_loop_map;
+}
+
+std::unordered_map<kir::ForLoop*, IterDomain*> loopToComputeAtMap(
+    TensorView* tv,
+    const std::vector<kir::ForLoop*>& loops,
+    const std::unordered_map<IterDomain*, IterDomain*>& ca_id_map) {
+  auto inverse_map = computeAtToLoopMap(tv, loops, ca_id_map);
+  std::unordered_map<kir::ForLoop*, IterDomain*> map;
+  for (auto entry : inverse_map) {
+    map[entry.second] = entry.first;
+  }
+  return map;
+}
+
+std::unordered_map<IterDomain*, IterDomain*> mapIdPtoC(
+    TensorView* producer,
+    TensorView* consumer) {
+  auto p2c_domain_ind_map = TensorDomain::mapDomainPandC(
+      producer->domain()->domain(), consumer->domain()->domain());
+
+  std::unordered_map<IterDomain*, IterDomain*> p2c_id_map;
+
+  for (auto entry : p2c_domain_ind_map) {
+    auto p_i = entry.first;
+    auto c_i = entry.second;
+    p2c_id_map[producer->getComputeAtAxis(p_i).first] =
+        consumer->getComputeAtAxis(c_i).first;
+  }
+
+  return p2c_id_map;
+}
+
+std::vector<Val*> getIndicesForTV(
+    TensorView* tv,
+    const std::vector<kir::ForLoop*>& loops,
+    bool for_predicates,
+    const std::unordered_map<IterDomain*, IterDomain*>& ca_id_map) {
+  Val* zero = new Int(0);
+  std::vector<Val*> indices(tv->nDims(), zero);
+
+  bool is_shared =
+      for_predicates ? false : tv->getMemoryType() == MemoryType::Shared;
+  bool is_local =
+      for_predicates ? false : tv->getMemoryType() == MemoryType::Local;
+
+  // Where is this TV allocated relative to the loop nest and its own axes?
+  auto alloc_point = loop_utils::getAllocPoint(tv, loops);
+  auto alloc_axis = alloc_point.second;
+
+  // Which loop is this axis associated with?
+  auto ca2loop = computeAtToLoopMap(tv, loops, ca_id_map);
+
+  bool need_reduction_axes = loopsHasReductions(tv, loops, ca_id_map);
+
+  // Look at each axis individually in out's domain
+  for (int64_t tv_i = 0; tv_i < (int64_t)tv->nDims(); tv_i++) {
+    if (tv->axis(tv_i)->isReduction() && !need_reduction_axes) {
+      continue;
+    }
+
+    // Check if we need to index based on this axis. If outside our allocation
+    // point, we don't need to unless we're generating a predicate
+    if (tv_i < alloc_axis && !for_predicates) {
+      continue;
+    }
+
+    auto ca_id = tv_i < tv->getThisComputeAtAxis()
+        ? tv->getComputeAtAxis(tv_i).first
+        : tv->axis(tv_i);
```

Comment:
Can this be just `tv->getComputeAtAxis(tv_i).first` always?

Reply:
I think so.